contracts/LockDealBundleProvider/LockDealBundleProvider.sol

```diff
@@ -35,29 +35,18 @@ contract LockDealBundleProvider is
         require(providerCount == providerParams.length, "providers and params length mismatch");
         require(providerCount > 1, "providers length must be greater than 1");
 
-        // create a new bundle pool owned by the owner with `0` token trasnfer amount
-        poolId = lockDealNFT.mint(owner, token, msg.sender, 0, address(this));
+        uint256 totalAmount = _calcTotalAmount(providerParams);
```

Comment:
Hmm... We iterate the same `for` statement twice. Is it fine?

Reply:
we need to know the sum when opening the main pool. I think its fine here.